core/lib/server/pub-filters.db.test.ts

```diff
@@ -0,0 +1,961 @@
+	},
+	{
+		title: "updatedAt & createdAt filters",
+		filter: {
+			$or: [
+				{
+					createdAt: {
+						$lte: new Date("2025-01-01"),
+					},
+				},
+				{
+					createdAt: {
+						$gte: new Date("2090-01-01"),
+					},
+				},
+			],
+		},
+		querystring:
+			"filters[$or][0][createdAt][$lte]=2025-01-01T00:00:00.000Z&filters[$or][1][createdAt][$gte]=2090-01-01T00:00:00.000Z",
+		sql: `"pubs"."createdAt" <= $1 or "pubs"."createdAt" >= $2`,
+		parameters: [
+			`"${new Date("2025-01-01").toISOString()}"`,
+			`"${new Date("2090-01-01").toISOString()}"`,
+		],
+		foundIds: [trueId, vector3Id],
+	},
+	{
+		title: "date filters",
+		filter: {
+			[slug("date")]: {
+				$eq: twenty99,
+			},
+		},
+		querystring: `filters[${slug("date")}][$eq]=${twenty99.toISOString()}`,
+		sql: `"slug" = $1 and "value" = $2`,
+		parameters: [slug("date"), `"${twenty99.toISOString()}"`],
+		foundIds: [numberArrayId, number24Id],
+	},
+	{
+		title: "multiple operators on same field should be treated as AND",
+		filter: {
+			[slug("number")]: { $gt: 40, $lt: 50 },
+		},
+		querystring: `filters[${slug("number")}][$gt]=40&filters[${slug("number")}][$lt]=50`,
+		sql: `"slug" = $1 and ("value" > $2 and "value" < $3)`,
+		parameters: [slug("number"), 40, 50],
+		foundIds: [number42Id],
+	},
+	{
+		title: "multiple fields should be treated as AND",
+		filter: {
+			[slug("title")]: { $eq: "Test" },
+			[slug("number")]: { $eq: 99 },
+		},
+		querystring: `filters[${slug("title")}][$eq]=Test&filters[${slug("number")}][$eq]=99`,
+		sql: /exists \(.*"slug" = \$1 and "value" = \$2\).* and exists \(.*"slug" = \$3 and "value" = \$4\).*/,
+		parameters: [slug("title"), '"Test"', slug("number"), "99"],
+		foundIds: [testTitleId],
+	},
+	{
+		title: "boolean coercion",
+		filter: {
+			[slug("boolean")]: { $eq: true },
+		},
+		querystring: `filters[${slug("boolean")}][$eq]=true`,
+		sql: `"slug" = $1 and "value" = $2`,
+		parameters: [slug("boolean"), "true"],
+		foundIds: [testCaseId, importantDocId, trueId],
+	},
+	{
+		title: "array in operator",
+		filter: {
+			[slug("number")]: { $in: [42, 24] },
+		},
+		querystring: `filters[${slug("number")}][$in][]=42&filters[${slug("number")}][$in][]=24`,
+		sql: `"slug" = $1 and "value" in ($2, $3)`,
+		parameters: [slug("number"), 42, 24],
+		foundIds: [number42Id, number24Id],
+	},
+	{
+		title: "between operator",
+		filter: {
+			[slug("number")]: { $between: [20, 25] },
+		},
+		querystring: `filters[${slug("number")}][$between][0]=20&filters[${slug("number")}][$between][1]=25`,
+		sql: `"slug" = $1 and ("value" >= $2 and "value" <= $3)`,
+		parameters: [slug("number"), 20, 25],
+		foundIds: [number24Id],
+	},
+	// {
+	// 	title: "null and notNull operators",
+	// 	filter: {
+	// 		[slug("relation")]: { $null: true },
+	// 	},
+	// 	querystring: `filters[${slug("relation")}][$null]`,
+	// 	sql: `"slug" = $1 and "value" is null`,
+	// 	parameters: [slug("relation")],
+	// 	foundIds: [relationId],
+	// },
```

Comment:
i couldn't really get these to work well, but they might also be superfluous: i added them in case we ever ended up setting values to `null` to indicate that the value is "empty", but we are likely not doing that right.